src/duk_debugger.c

```diff
@@ -1639,6 +1684,47 @@ DUK_LOCAL void duk__debug_process_message(duk_hthread *thr) {
 	return;
 }
 
+/* Halt execution and enter a debugger message loop until execution is resumed
+ * by the client. PC for the current activation may be temporarily decremented
+ * so that the "current" instruction will be shown by the client.
+ */
+
+DUK_INTERNAL void duk_debug_halt_execution(duk_hthread *thr, duk_bool_t use_prev_pc) {
+	duk_activation *act;
+	duk_hcompiledfunction *fun;
+	duk_instr_t *old_pc;
+
+	DUK_HEAP_SET_PAUSED(thr->heap);
+
+	act = duk_hthread_get_current_activation(thr);
+	fun = (duk_hcompiledfunction *) DUK_ACT_GET_FUNC(act);
```

Comment:
Results of testing:
* There is no issue when an error is thrown from a Duktape/C function.  There is no actual compiledfunction in that case, but as there is a check for that, it's completely safe.
* The actual, concrete issue is this: If an error is thrown with _nothing_ on the Duktape callstack (as is the case for, say, a top-level `duk_call()` throwing a SyntaxError), it will segfault because `duk_hthread_get_current_activation(thr);` returns NULL.